File: ProgLangChallenge/Java/CSVReader.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.*; 

public class CSVReader
{
    private String fileName;
    private List<String> voter; 
    private List<List<String>> candidates;  
    public CSVReader()
    { 
        fileName = ""; 
        voter = new ArrayList<String>(); 
        candidates = new ArrayList<List<String>>();
    }

    public void setFileName(String fileName)
    {
        this.fileName = fileName; 
    }

    public String getFileName()
    {
        return fileName; 
    } 

    public void readFile(String fileName)
    { 
        String line = ""; 
        String csvSplitBy= ","; 
        int counter = 0; 
        try(BufferedReader br = new BufferedReader(new FileReader(fileName)))
        {
            while((line = br.readLine()) != null)
            {
                String[] values =  line.split(csvSplitBy);
                if(counter == 0)
                {
                    voter = Arrays.asList(values);
                }
                else 
                {
                    candidates.add(Arrays.asList(values));
                }
                counter++;
                
            }
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
    }

    public List<String> getVoter()
    {
        return voter;
    }

    public List<List<String>> getCandidates()
    {
        return candidates;
    }
}

File: ProgLangChallenge/C++/CSVReader.cpp
#include "CSVReader.h"
#include <iostream> 
#include <vector> 
#include <sstream>
#include <fstream>

using namespace std; 

void CSVReader::readFile(string fileName)
{
    ifstream  data(fileName);
    string line;
    int counter = 0; 
    while(getline(data,line))
    {
        stringstream lineStream(line);
        string cell;
        vector<string> parsedRow;
        while(getline(lineStream,cell,','))
        {
            parsedRow.push_back(cell);
        }
        if(counter == 0)
        {
            voter = parsedRow;
        }
        else
        { 
            candidates.push_back(parsedRow);
        }
        counter++;
        
    }
}

void CSVReader::print1DVector(vector<string> array)
{
    for(int i = 0; i < array.size(); i++)
    {
        cout << array.at(i) << " ";
    }
    cout << endl; 
}

void CSVReader::print2DVector(vector<vector<string>> array2d)
{
    for (int i = 0; i < array2d.size(); i++)
    {
        print1DVector(array2d[i]);
    }   
}



